### PR TITLE
Simplify our dependency creation by using plugin

### PR DESCRIPTION
**Pull request by rogerhu**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ plugins {
     id "com.jfrog.artifactory" version "4.4.18"
     id "com.jfrog.bintray" version "1.7.3"
     id 'com.github.ben-manes.versions' version '0.17.0'
+    id 'digital.wup.android-maven-publish' version '3.0.0' // tweak depending on Gradle version
```
Comment (rogerhu) [inline build.gradle]: this plugin version depends what Gradle version is used.  The api for gradle keeps changing..

**Comment by Jawnnypoo** [comment]
I have used this one in the past, not sure what the differences are.

https://github.com/dcendents/android-maven-gradle-plugin

But this looks good 👍

**Comment by rogerhu** [comment]
There are two Maven plugins, maven and maven-publish.  The one you've used (and I have done so myself) uses the maven plugin.

  Bintray's official upload script relies on the maven-publish plugin.  But it fails to produce dependencies in the metadata for Android.  This plugin takes care of that part.

**Comment by Jawnnypoo** [comment]
Ahh okay, makes sense. Any ideas about the build failing? Seems like an odd error.

**Comment by rogerhu** [comment]
I bumped

**Pull request merged**